system-addon/lib/PrefsFeed.jsm

```diff
@@ -17,12 +18,17 @@ this.PrefsFeed = class PrefsFeed {
   constructor(prefMap) {
     this._prefMap = prefMap;
     this._prefs = new Prefs();
+    this._storage = new ActivityStreamStorage("sectionPrefs");
   }
 
   // If the any prefs are set to something other than what the prerendered version
   // of AS expects, we can't use it.
-  _setPrerenderPref(name) {
-    this._prefs.set("prerender", PrerenderData.arePrefsValid(pref => this._prefs.get(pref)));
+  async _setPrerenderPref() {
+    if (!this._storage.intialized) {
+      await this._storage.init();
```

Comment:
initializing the same store in two places could produce a race condition where `storage.init()` is called a second time while the first one is still finishing resolving – maybe use a promise that can be awaited or cache a reference to the initialization as a promise?

Reply:
There might not be a good reason for calling `.init` twice. Need to think about it but when the preference feed is initialized we call `_setPrerenderPref` I should just call it once in `PrefsFeed.init`.
If both are required I'll cache the reference to init as you suggested.